lib/order.rb

```diff
@@ -1,18 +1,101 @@
+##### Qs: what is the purpose of the id?
+# should there be an array of hashes to store what the store has already?
+# should I initialize price (Why only id and products?)
+# how to add to product list?
+require 'csv'
 module Grocery
   class Order
     attr_reader :id, :products
+    @@all_orders = []
 
     def initialize(id, products)
       @id = id
       @products = products
-    end
+
+    end # end of initialize
+    # zero products is permitted
+    # you can assume that there is only one of each product
 
     def total
-      # TODO: implement total
-    end
+      add_product = 0
+      @products.each_value do |cost|
+        add_product  += cost
+      end
+      total = (add_product + (add_product * 0.075))
+      return total.round(2)
+    end # total method
+
+    # @products.each do |product, values|
+    #   sum = @products.values.inject(0, :+)
+    #   return (sum + (sum * 0.075).round(2))
+    # end # products loop
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      if @products.key?(product_name) == true
+        return false
+      elsif
+        @products[product_name] = product_price
+        return true
+      end # conditional
+    end # add_product method
+
+    def remove_product(product_name)
+      if @products.key?(product_name) == true
+        @products.delete(product_name)
+        return true
+      else
+        return false
+      end
     end
-  end
-end
+
+    def self.all
+      if @@all_orders.length > 0
+        return @@all_orders
+      end
```

Comment:
I like this - if we've already loaded the orders once, why do it again.

This technique is called "memoization", and it's a form of caching. There's potentially a big performance benefit here, because reading from disk is extremely expensive, but the downside is we don't have any way to pick up changes to the file made by a user or another program.